### DISCUSSION TITLE
Is the `Client` pooled?

### DISCUSSION THREAD

**Discussion by feelingsonice** [nats-io/nats.rs#1311] [Q&A] [answered]
Does the `Client` implement connection pooling? Should I put try and wrap it with something like [`deadpool`](https://www.google.com/search?client=safari&rls=en&q=deadpool+rust&ie=UTF-8) or is that unnecessary?

**Comment by Jarema** [comment, marked as answer]
Client does not implement connection pooling, as NATS is really good at head of line blocking and client normally should not require more than one connection.
